internal/handlers/pg/msg_update.go

```diff
@@ -67,18 +66,17 @@ func (h *Handler) MsgUpdate(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg,
 		return nil, err
 	}
 
-	created, err := pgdb.CreateCollectionIfNotExist(ctx, h.PgPool, sp.DB, sp.Collection)
+	err = h.PgPool.InTransactionRetry(ctx, func(tx pgx.Tx) error {
+		return pgdb.CreateCollectionIfNotExists(ctx, tx, sp.DB, sp.Collection)
+	})
 	if err != nil {
-		if errors.Is(pgdb.ErrInvalidTableName, err) ||
+		if errors.Is(pgdb.ErrInvalidCollectionName, err) ||
```

Comment:
I know it was already there, but:
```
func errors.Is(err error, target error) bool

Is reports whether any error in err's chain matches target.
...
```

Shouldn't we use a different order of arguments here?

Reply:
hehehe, nice catch! (and another reason why we need to cover such things with tests)

Reply:
fixed